components/dns/main.tf

```diff
@@ -0,0 +1,21 @@
+data "local_file" "configuration" {
+  filename = "${path.cwd}/../../apps.yaml"
+}
+
+locals {
+  apps = yamldecode(data.local_file.configuration.content).apps
+  cname_records = flatten([
+    for app in local.apps : split(".", replace("${app.externalUrl}", "https://", ""))[0]
+  ])
+}
+
+
+resource "azurerm_dns_cname_record" "this" {
+  for_each = toset(local.cname_records)
+
+  name                = each.value
+  zone_name           = "hmcts.net"
```

Comment:
this assumes the only zone we use is hmcts.net is that safe? should it extract the zone?

Reply:
Pushed, a bit hacky but works